doc/install/configure_ide.rst

```diff
@@ -0,0 +1,48 @@
+.. _configure-ide:
+
+IDE Configuration Guide
+=======================
+
+.. include:: ../links.inc
+.. highlight:: console
+
+Thanks for taking the time to contribute! This guide steps through various IDE settings that are helpful when
+contributing to MNE-Python.
+
+Some of the basic settings include enabling linting, setting line margins, introspection, and docstring type:
+
+* Linting - is the automated checking of your source code for programmatic and stylistic errors. Most of the time basic linting is already enabled but let's double check.
+
+* Line margins - we have code style rules that forbid lines longer than 79 characters, so having a ruler at 79 characters helps coders adhere to that rule.
+
+* Introspection - this helps removes trailing whitespaces that could be problematic when compiling or building
+
+* Docstring - we use NumPy because it supports a combination of reStructured and GoogleDocstrings and supported by Sphinx
+
+
+Spyder
+------
+
+* :menuselection:`Preferences --> Completion and Linting --> Linting --> Enable Basic Linting`
+* :menuselection:`Preferences --> Completion and Linting --> Introspection --> Advanced -->Add "mne" to list of modules to preload for code completion`
+* :menuselection:`Preferences --> Editor --> Show Vertical Lines at == 79`
+* :menuselection:`Preferences --> Editor --> Advanced Settings --> Docstring Type == Numpy`
+* :menuselection:`Preferences --> Editor --> Code Introspection/Analysis and “Automatically remove trailing spaces when saving files”`
```

Comment:
A couple of modification/additions, to match the entries in the latest Spyder 5.4.2.

```
:menuselection:`Preferences --> Editor --> Show Vertical Lines at == 79`
```

This entry has been moved to Completion and linting -> Code style and formatting -> Line length.
In the same section, we could recommend to change the code formatting provider from `autopep8` to `black`.

In Completion and linting -> Docstring style -> Enable docstring style linting and choose the `numpy` convention replaces:
```
:menuselection:`Preferences --> Editor --> Advanced Settings --> Docstring Type == Numpy
```

And finally:

```
:menuselection:`Preferences --> Editor --> Code Introspection/Analysis and “Automatically remove trailing spaces when saving files”
```

is replaced by Editor -> Source code -> Automatically remove trailing spaces when saving files.
In the same tab, we could also recommend to enable: 
- Insert a newline at the end if one does not exist when saving a file
- Trim all newlines after the final one when saving a file

And in the Advaned settings tab -> Convert end-of-line character to the following on save -> LF (Unix).

Reply:
This actually makes me think that this will be very hard to maintain / keep up to date ... Would it be enough to only keep the general recommendations but not the "how-to" regarding the IDE's?

Reply:
Exactly what I thought when I noticed all the differences..